packages/core/src/data-access/documentVersions.ts

```diff
@@ -3,21 +3,39 @@ import {
   database,
   DocumentVersion,
   documentVersions,
-  findCommit,
-  getCommitMergedAt,
+  findCommitById,
   Result,
   TypedResult,
 } from '@latitude-data/core'
 import { LatitudeError, NotFoundError } from '$core/lib/errors'
 import { and, eq, isNotNull, lte, max } from 'drizzle-orm'
 
+async function getCommitMergedAt({
+  commitId,
+}: {
+  commitId: number
+}): Promise<TypedResult<Date | null, LatitudeError>> {
+  const commitResult = await findCommitById({ id: commitId })
+  if (commitResult.error) return commitResult
+
+  return Result.ok(commitResult.value.mergedAt)
+}
+
 export async function getDocumentsAtCommit(
-  { commitUuid, projectId }: { commitUuid: string; projectId: number },
+  {
+    commitId,
+    commitMergedAt,
```

Comment:
this api is unintuitive as it accepts both a commitId for a commit and another merged_at property which also belongs to commit. 

You could split it into getDocumentsAtDate and getDocumentsAtCommit methods, for example.

Reply:
I only added `mergedAt` for performance, as it is needed for retrieving the documents, and several parts of the code that use this function already have this information so I didn't want to fetch this if it's not necessary. But I agree that it's unintuitive.